std-bits/table/src/main/java/org/enso/table/data/column/storage/DoubleStorage.java

```diff
@@ -357,4 +368,46 @@ public Storage<Double> slice(List<SliceRange> ranges) {
 
     return new DoubleStorage(newData, newSize, newMissing);
   }
+
+  @Override
+  public Storage<?> cast(StorageType targetType, CastProblemBuilder castProblemBuilder) {
+    return switch (targetType) {
+      case AnyObjectType any -> new MixedStorageFacade(this);
+      case FloatType floatType -> this;
+      case IntegerType integerType -> {
+        int n = size();
+        NumericBuilder builder = NumericBuilder.createLongBuilder(n);
+        double min = (double) integerType.getMinValue();
+        double max = (double) integerType.getMaxValue();
+        for (int i = 0; i < n; i++) {
+          if (isMissing.get(i)) {
+            builder.appendNulls(1);
+          } else {
+            double value = getItem(i);
+            if (value < min || value > max) {
+              builder.appendNulls(1);
+              castProblemBuilder.reportLossyConversion();
```

Comment:
`Lossy` feels like the wrong name here - this is a failed conversion as out of range.
This would change my previous comments on "lossy" stuff on date/datetime.